File: leetcode/1.py
#Solution 1
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        result = []

        for index, number in enumerate(nums):
            difference = target - number

            if difference in nums[index+1]:
                result.append(index)
                result.append(nums[index+1:].index(difference)+index+1)

        return result

File: Java/1.java
class Solution:{
    public int[] twoSum(int[] nums, int target){
        /*
         * 1. create a hashmap to store numbers in nums
         * 2. using for loop to compare the difference between target and difference of two numbers
         * 3. if matches then return indices of them
         */

        
        HashMap<Integer, Integer> hash = new HashMap<Integer,Integer>();

        for (int i = 0, i < nums.length; i++){
            int diff = target - nums[i];

            if (hash.containsKey(diff)){
                return new int[] {hash.get(diff)};
            }

            hash.put(nums[i], i);
        }
        return new int[] {};

    }
}